arangod/Agency/State.cpp

```diff
@@ -1033,25 +1064,17 @@ bool State::loadRemaining() {
           term_t term(ii.get("term").getNumber<uint64_t>());
           for (index_t i = lastIndex + 1; i < index; ++i) {
             LOG_TOPIC(WARN, Logger::AGENCY) << "Missing index " << i << " in RAFT log.";
-            _log.push_back(log_t(i, term, buf, std::string()));
+            _log.emplace_back(log_t(i, term, buf, std::string()));
             lastIndex = i;
           }
           // After this loop, index will be lastIndex + 1
         }
 
         if (index == lastIndex + 1 || (index == lastIndex && _log.empty())) {
           // Real entries
-          try {
-            _log.push_back(log_t(basics::StringUtils::uint64(
-                                     ii.get(StaticStrings::KeyString).copyString()),
-                                 ii.get("term").getNumber<uint64_t>(), tmp, clientId));
-          } catch (std::exception const& e) {
-            LOG_TOPIC(ERR, Logger::AGENCY)
-                << "Failed to convert " + ii.get(StaticStrings::KeyString).copyString() +
-                       " to integer."
-                << e.what();
-          }
-
+          logEmplaceBackNoLock(
```

Comment:
If you really want to emplace here you would have to pass in the pieces by pieces and then use `emplace` inside with no `log_t` constructor, as exlained above.